internal/validate/helpers.go

```diff
@@ -55,6 +55,32 @@ func GetPolicyConfig(ctx context.Context, policyConfiguration string) (string, e
 		policyConfiguration = configFile
 	}
 
+	// Check if policyConfig is a url. If so, try to download it. If successful we write
+	// that into that into the data.policyConfiguration var.
+	if source.SourceIsHttps(policyConfiguration) {
```

Comment:
Fair bit of duplication with the code above, perhaps setting a variable to the download function based on those conditionals could help reduce the duplication, something like:

```go
var downloadFn func(context.Context, string, string)
if source.SourceIsGit(policyConfiguration) {
	downloadFn = source.GitConfigDownload
} else if source.SourceIsHttps(policyConfiguration) {
	downloadFn = source.HttpsConfigDownload
}

if downloadFn == nil {
	return "", nil // or something?
}

fs := utils.FS(ctx)
tmpDir, err := utils.CreateWorkDir(fs)
if err != nil {
	return "", err
}
defer utils.CleanupWorkDir(fs, tmpDir)

// Https download and find a suitable config file
configFile, err := downloadFn(ctx, tmpDir, policyConfiguration)
if err != nil {
	return "", err
}
// ...
```